cyberdrop_dl/scraper/crawlers/nekohouse_crawler.py

```diff
@@ -0,0 +1,237 @@
+            await self.profile(scrape_item)
+        else:
+            await self.handle_direct_link(scrape_item)
+
+        await self.scraping_progress.remove_task(task_id)
+
+    @error_handling_wrapper
+    async def profile(self, scrape_item: ScrapeItem) -> None:
+        """Scrapes a profile"""
+        soup = await self.client.get_BS4(self.domain, scrape_item.url)
+        offset, maximum_offset = await self.get_offsets(scrape_item, soup)
+        service, user = await self.get_service_and_user(scrape_item)
+        user_str = await self.get_user_str_from_profile(soup)
+        service_call = self.primary_base_domain / service / "user" / user
+        while offset <= maximum_offset:
+            async with self.request_limiter:
+                soup: BeautifulSoup = await self.client.get_BS4(self.domain, service_call.with_query({"o": offset}))
+                offset += 50
+                
+                posts = soup.select(self.post_selector)
+                if not posts:
+                    break
+                for post in posts:
+                    # Create a new scrape item for each post
+                    post_id = post.get("href", "").split("/")[-1]
+                    if post_id == "":
+                        continue
+                    post_url = self.primary_base_domain / service / "user" / user / "post" / post_id
+                    # Call on self.post to scrape the post by creating a new scrape item
+                    new_scrape_item = await self.create_scrape_item(scrape_item, post_url, "", add_parent=self.primary_base_domain / service / "user" / user)
+                    await self.post(new_scrape_item)
+
+    @error_handling_wrapper
+    async def post(self, scrape_item: ScrapeItem) -> None:
+        """Scrapes a post"""
+        service, user, post_id = await self.get_service_user_and_post(scrape_item)
+        await self.get_post_content(scrape_item, post_id, user, service)
+
+    @error_handling_wrapper
+    async def get_post_content(self, scrape_item: ScrapeItem, post: int, user: str, service: str) -> None:
+        """Gets the content of a post and handles collected links"""
+        if post == 0:
+            return
+
+        post_url = self.primary_base_domain / service / "user" / user / "post" / str(post)
+        async with self.request_limiter:
+            soup: BeautifulSoup = await self.client.get_BS4(self.domain, post_url)
+            data = {
+                "id": post,
+                "user": user,
+                "service": service,
+                "title": soup.select_one(self.post_title_selector).text.strip(),
+                "content": soup.select_one(self.post_content_selector).text.strip(),
+                "user_str": soup.select_one(self.post_author_username_selector).text.strip(),
+                "published": soup.select_one(self.post_timestamp_selector).text.strip(),
+                "file": [],
+                "attachments": []
```

Comment:
Everything in this dict that comes from the soup should be on a different dict and inside `try` / `except` block for `AtributeError` cause they are necessary. Then re-raise as `ScrapeFailure`